src/test/java/teammates/test/cases/util/StringHelperTest.java

```diff
@@ -406,19 +407,36 @@ public void testRemoveNonAscii() {
 
     @Test
     public void testJoin() {
-        assertEquals("", StringHelper.join("", new String[] {}));
-        assertEquals("", StringHelper.join(",", new String[] {}));
-        assertEquals("", StringHelper.join("||", new String[] {}));
+        assertEquals("", StringHelper.join(""));
+        assertEquals("", StringHelper.join(","));
+        assertEquals("", StringHelper.join("||"));
 
-        assertEquals("test", StringHelper.join("", new String[] {"test"}));
-        assertEquals("test", StringHelper.join(",", new String[] {"test"}));
-        assertEquals("test", StringHelper.join("||", new String[] {"test"}));
-        assertEquals("testdata", StringHelper.join("", new String[] {"test", "data"}));
+        assertEquals("test", StringHelper.join("", "test"));
+        assertEquals("test", StringHelper.join(",", "test"));
+        assertEquals("test", StringHelper.join("||", "test"));
+        assertEquals("testdata", StringHelper.join("", "test", "data"));
 
-        assertEquals("test,data", StringHelper.join(",", new String[] {"test", "data"}));
-        assertEquals("test||data", StringHelper.join("||", new String[] {"test", "data"}));
+        assertEquals("test,data", StringHelper.join(",", "test", "data"));
+        assertEquals("test||data", StringHelper.join("||", "test", "data"));
         assertEquals("test|||data|||testdata",
-                StringHelper.join("|||", new String[] {"test", "data", "testdata"}));
+                StringHelper.join("|||", "test", "data", "testdata"));
+    }
+
+    @Test
+    public void testJoinWithListOfIntegers() {
+        assertEquals("", StringHelper.join(",", new ArrayList<Integer>()));
+        assertEquals("5,14", StringHelper.join(",", Arrays.asList(5, 14)));
+        assertEquals("5||14", StringHelper.join("||", Arrays.asList(5, 14)));
```

Comment:
Just a quick comment in passing: interesting test cases are 0 elements, 1 element, 2 elements. you are missing the '1 element' case.